tracer/src/Datadog.Trace/IAST/Aspects/StringAspect.cs

```diff
@@ -0,0 +1,244 @@
+// <copyright file="StringAspect.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Iast.Dataflow;
+
+namespace Datadog.Trace.Iast.Aspects
+{
+    /// <summary> String class aspects </summary>
+    [AspectClass("mscorlib,netstandard,System.Private.CoreLib", AspectFilter.StringOptimization)]
+    [System.ComponentModel.Browsable(false)]
+    [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
+    public partial class StringAspect
+    {
+        /// <summary>
+        /// Concat with both strings not static
+        /// </summary>
+        /// <param name="target">main string instance (not static)</param>
+        /// <param name="param1">first parameter (not static)</param>
+        /// <returns>target concatenated to param1</returns>
+        [AspectMethodReplace("System.String::Concat(System.String,System.String)", AspectFilter.StringLiterals_Any)]
+        public static string Concat(string target, string param1)
+        {
+            return Concat_Internal(new string[] { target, param1 });
+        }
+
+        /// <summary>
+        /// Concat with param1 not static
+        /// </summary>
+        /// <param name="target">main string instance</param>
+        /// <param name="param1">first parameter (not static)</param>
+        /// <returns>target concatenated to param1</returns>
+        [AspectMethodReplace("System.String::Concat(System.String,System.String)", AspectFilter.StringLiteral_0)]
+        public static string Concat_0(string target, string param1)
+        {
+            return Concat_Internal(new string[] { target, param1 }, 0);
+        }
+
+        /// <summary>
+        /// Concat with instance strings not static
+        /// </summary>
+        /// <param name="target">main string instance (not static)</param>
+        /// <param name="param1">first parameter</param>
+        /// <returns>target concatenated to param1</returns>
+        [AspectMethodReplace("System.String::Concat(System.String,System.String)", AspectFilter.StringLiteral_1)]
+        public static string Concat_1(string target, string param1)
+        {
+            return Concat_Internal(new string[] { target, param1 }, 1);
+        }
+
+        /// <summary>
+        /// String Concat with object
+        /// </summary>
+        /// <param name="target">main string instance</param>
+        /// <param name="param1">first parameter</param>
+        /// <returns>target concatenated to param1</returns>
+        [AspectMethodReplace("System.String::Concat(System.Object,System.Object)")]
```

Comment:
A question: Can filters like AspectFilter.StringLiteral_1 be used when parameters are defined as object?

Reply:
Yes, it just checks the param is not a static string